spatialdata/_io/write.py

```diff
@@ -66,133 +70,231 @@ def _write_metadata(
-    chunks: Optional[Union[Tuple[Any, ...], int]] = None,
+    name: str,
+    group_type: str = "ngff:polygons",
+    storage_options: Optional[Union[JSONDict, List[JSONDict]]] = None,
     fmt: Format = SpatialDataFormat(),
     axes: Optional[Union[str, List[str], List[Dict[str, str]]]] = None,
-    coordinate_transformations: Optional[List[List[Dict[str, Any]]]] = None,
-    storage_options: Optional[Union[JSONDict, List[JSONDict]]] = None,
     **metadata: Union[str, JSONDict, List[JSONDict]],
 ) -> None:
-    write_image_ngff(
-        image=image,
-        group=group,
-        scaler=scaler,
-        chunks=chunks,
+    sub_group = group.require_group("polygons")
+    polygons_groups = sub_group.require_group(name)
+    coordinate_transformations = [[polygons.attrs.get("transform").to_dict()]]
+    print(coordinate_transformations)
+
+    # TODO: save everything or just polygons?
+    geometry, coords, offsets = to_ragged_array(polygons.geometry)
+    polygons_groups.create_dataset(name="coords", data=coords)
+    polygons_groups.create_dataset(name="offsets", data=offsets)
+    # polygons_groups = sub_group[name]
+    attr = {"geos": {"geometry_name": geometry.name, "geometry_type": geometry.value}}
+
+    _write_metadata(
+        polygons_groups,
+        group_type=group_type,
+        shape=coords.shape,
+        attr=attr,  # type: ignore[arg-type]
         fmt=fmt,
         axes=axes,
         coordinate_transformations=coordinate_transformations,
-        storage_options=storage_options,
         **metadata,
     )
 
 
-def write_labels(
-    labels: ArrayLike,
+def write_shapes(
+    shapes: AnnData,
     group: zarr.Group,
     name: str,
-    scaler: Scaler = Scaler(),
-    chunks: Optional[Union[Tuple[Any, ...], int]] = None,
+    group_type: str = "ngff:shapes",
     fmt: Format = SpatialDataFormat(),
     axes: Optional[Union[str, List[str], List[Dict[str, str]]]] = None,
-    coordinate_transformations: Optional[List[List[Dict[str, Any]]]] = None,
-    storage_options: Optional[Union[JSONDict, List[JSONDict]]] = None,
-    label_metadata: Optional[JSONDict] = None,
-    **metadata: JSONDict,
+    **metadata: Union[str, JSONDict, List[JSONDict]],
 ) -> None:
-    if np.prod(labels.shape) == 0:
-        # TODO: temporary workaround, report bug to handle empty shapes
-        # TODO: consider the different axes, now assuming a 2D image
-        coordinate_transformations = fmt.generate_coordinate_transformations(shapes=[(1, 1)])
-    write_labels_ngff(
-        labels=labels,
-        group=group,
-        name=name,
-        scaler=scaler,
-        chunks=chunks,
+    sub_group = group.require_group("shapes")
+    coordinate_transformations = [[shapes.uns.get("transform").to_dict()]]
+    attr = shapes.uns.get("spatialdata_attrs")
+    # TODO: do not save copy but pop transform?
+    shapes_copy = shapes.copy()
+    shapes_copy.uns.pop("transform")
+    write_adata(sub_group, name, shapes_copy)
```

Comment:
basically if `adata.uns["transform"] = spatialdata.Transform(...)` it can't ofc be serialized so it needs to be popped. However, that would change the in-memory representation. Can only copy or someone has better idea?